napari/_qt/layer_controls/qt_image_controls.py

```diff
@@ -52,10 +57,17 @@ class QtImageControls(QtBaseImageControls):
         Label for the rendering mode dropdown menu.
     """
 
+    layer: 'napari.layers.Image'
```

Comment:
love the type hints thank you.  Can you put all of these type hints on the parameter itself though?  There's a subtle difference, which is that when you do it like this there's no error if you pass in the wrong type:

<img width="294" alt="Screen Shot 2022-06-26 at 6 33 12 AM" src="https://user-images.githubusercontent.com/1609449/175810166-3e5f90c9-31ad-4583-a76e-1ea1dc22a2f1.png">

<img width="336" alt="Screen Shot 2022-06-26 at 6 33 38 AM" src="https://user-images.githubusercontent.com/1609449/175810165-13d9c7e5-a4af-4b9e-af88-b76c54286428.png">

Reply:
I could put it in the constructor signature too, but cannot remove this without playing with generics. Because in your example there is a direct assignment to a variable and in this code part the assignment is done in the super class constructor. So annotating only constructor parameter will not help with finding all usage that is required during code refactoring. 

Especially, in this case constructor parameter is annotated in docstring, so PyCharm will highlight that the wrong type passed, but still will not know that the `.layer` attribute is the `Image` type.

Reply:
ok fair enough :)  (I don't mind generics fwiw, but agreed it's a lot more)